### PR TITLE
[DEVOPS-940] add haddock documentation hydra artifact

### PR DESCRIPTION
**Pull request by disassembler**
Adds haddock documentation to hydra

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -21,10 +21,13 @@ let
...
     testTarget = "--show-details=streaming";
   });
+  addHaddockDocumentation = drv: overrideCabal drv (attrs: {
+    doHaddock = true;
+  });
   plutusPkgs = (import ./pkgs { inherit pkgs; }).override {
     overrides = self: super: {
       plutus-prototype = addRealTimeTestLogs super.plutus-prototype;
-      language-plutus-core = addRealTimeTestLogs super.language-plutus-core;
+      language-plutus-core = addHaddockDocumentation (addRealTimeTestLogs super.language-plutus-core);
```
Comment (michaelpj) [inline default.nix]: `pkgs.haskell.lib.doHaddock`?

**Comment by mchakravarty** [comment]
I agree with @rvl about the depth of the docs tar, but otherwise, this looks great. Thanks!

File: default.nix [hunk 2]
```diff
@@ -18,14 +18,23 @@ with pkgs.lib;
...
 
 let
+  doHaddockHydra = drv: overrideCabal drv (attrs: {
+    doHaddock = true;
+    postInstall = ''
+      ${attrs.postInstall or ""}
+      mkdir -pv $doc/nix-support
+      tar -czvf $doc/${attrs.pname}-docs.tar.gz -C $doc/share/doc/html .
+      echo "file binary-dist $doc/${attrs.pname}-docs.tar.gz" >> $doc/nix-support/hydra-build-products
+      echo "report ${attrs.pname}-docs.html $doc/share/doc/html index.html" >> $doc/nix-support/hydra-build-products
```
Comment (michaelpj) [inline default.nix]: Do we need both this and the `docs/language-plutus-core.nix` below? The latter seems unnecessary to me, unless I'm missing something.

Comment (disassembler) [reply default.nix]: yeah, I missed that when cleaning up. It's not even referenced in default.nix anymore.

**Pull request merged**